core/test/surplus-test.ts

```diff
@@ -0,0 +1,105 @@
+import { expect } from 'chai';
+import {
+    fetchActiveSurplusAuctions,
+    bidToSurplusAuction,
+    collectSurplusAuction,
+    fetchSurplusAuctionByIndex,
+    restartSurplusAuction,
+} from '../src/surplus';
+import { setAllowanceAmountMKR } from '../src/authorizations';
+import getSigner, { setSigner, createSigner } from '../src/signer';
+import { setupRpcUrlAndGetNetworks } from '../src/rpc';
+import { swapToMKR } from '../src/helpers/swap';
+
+import { SurplusAuctionActive } from '../src/types';
+import hre from 'hardhat';
+
+const REMOTE_RPC_URL = process.env.REMOTE_RPC_URL;
+const HARDHAT_PRIVATE_KEY = '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80'; // deterministic private key from hardhat.
+const HARDHAT_FORK_BLOCK_NUMBER = 14078339;
+
+async function createWalletFromPrivateKey(privateKey: string, network: string) {
+    setSigner(network, createSigner(network, privateKey));
+    const signer = await getSigner(network);
+    return await signer.getAddress();
+}
```

Comment:
This can be moved to `signer.ts` if you find it useful, so a) you will need to import only one thing b) it can be reused later on